client/src/components/dataset/sidebars/ChangeDataType.jsx

```diff
@@ -0,0 +1,207 @@
+        type="text"
+        value={defaultValue}
+        onChange={(event) => onChange(event.target.value)}
+      />
+    </div>
+  );
+}
+
+DefaultValueInput.propTypes = {
+  defaultValue: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+};
+
+
+const errorOptions = [
+  {
+    value: 'empty-cell',
+    label: 'Leave empty',
+  },
+  {
+    value: 'default-value',
+    label: 'Pick a default value',
+  },
+  {
+    value: 'fail',
+    label: 'Let the transform fail',
+  },
+  {
+    value: 'delete-row',
+    label: 'Delete the row',
+  },
+];
+
+export default class ChangeDataType extends Component {
+
+  constructor() {
+    super();
+    this.state = {};
+    this.handleChangeErrorStrategy = this.handleChangeErrorStrategy.bind(this);
+    this.mergeArgs = this.mergeArgs.bind(this);
+  }
+
+
+  componentWillMount() {
+    const { column, newColumnType } = this.props;
+    this.setState({
+      op: 'core/change-datatype',
+      args: {
+        columnName: column.columnName,
+        newType: newColumnType,
+        defaultValue: null,
+        dateFormat: 'YYYY-MM-DD',
+      },
+      onError: 'default-value',
+      errorStrategy: 'empty-cell',
+    });
+  }
+
+
+  mergeArgs(args) {
+    this.setState({ args: Object.assign({}, this.state.args, args) });
+  }
+
+  handleChangeErrorStrategy(errorStrategy) {
+    switch (errorStrategy) {
+      case 'emptyCell': {
+        const args = Object.assign({}, this.state.args, { defaultValue: null });
+        this.setState({
+          args,
+          onError: 'default-value',
+          errorStrategy,
+        });
+        break;
+      }
+      case 'default-value':
+      case 'fail':
+      case 'delete-row': {
+        this.setState({
+          onError: errorStrategy,
+          errorStrategy,
+        });
+        break;
+      }
+      default:
+        throw new Error(`Unkown error strategy ${errorStrategy}`);
+    }
+  }
+
+  render() {
+    const { column, dataTypeOptions, onClose, onApply } = this.props;
+    const { newType } = this.state.args;
+    return (
+      <div
+        className="DataTableSidebar"
+        style={{
+          width: '300px',
+          height: 'calc(100vh - 4rem)',
+        }}
+      >
+        <SidebarHeader onClose={onClose}>
+          Change data type for {column.title}
+        </SidebarHeader>
+        <div className="inputs">
+          <div className="inputGroup">
+            <label htmlFor="dataTypeMenu">
+              Change data type to:
+            </label>
+            <DashSelect
+              name="dataTypeMenu"
+              value={newType}
+              options={dataTypeOptions}
+              onChange={type => {
+                if (type === 'date') {
+                  this.mergeArgs({ newType: 'date', dateFormat: 'YYYY-MM-DD' });
+                }
+                this.mergeArgs({ newType: type });
+              }}
+            />
+          </div>
+          {newType === 'date' && column.type === 'text' ?
```

Comment:
Yeah, it's a hint on how to parse the text. We store dates as timestamps
internally

On Mon, May 30, 2016 at 4:25 PM Gabriel Martin <notifications@github.com>
wrote:

> In client/src/components/dataset/sidebars/ChangeDataType.jsx
> <https://github.com/akvo/akvo-lumen/pull/182#discussion_r65071509>:
>
> > +            <label htmlFor="dataTypeMenu">
> > +              Change data type to:
> > +            </label>
> > +            <DashSelect
> > +              name="dataTypeMenu"
> > +              value={newType}
> > +              options={dataTypeOptions}
> > +              onChange={type => {
> > +                if (type === 'date') {
> > +                  this.mergeArgs({ newType: 'date', dateFormat: 'YYYY-MM-DD' });
> > +                }
> > +                this.mergeArgs({ newType: type });
> > +              }}
> > +            />
> > +          </div>
> > +          {newType === 'date' && column.type === 'text' ?
>
> Ah I see - turns out I misunderstood what the date format option was for,
> I was viewing it as a display option rather than a parsing option. I can
> now see why this makes sense
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/akvo/akvo-lumen/pull/182/files/e39cb39d3f6a6682b3540de14959d8a0fe6b7dbb#r65071509>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAGKn3JmIX9YZBUvD9-NVGLNggytBx1oks5qGuVNgaJpZM4IpjX6>
> .
>